### PR TITLE
[tlse] TLS database connection

### PR DESCRIPTION
**Pull request by stuggi**
The my.cnf file gets added to the secret holding the service configs. The content of my.cnf is centrally managed in the mariadb-operator and retrieved calling db.GetDatabaseClientConfig(tlsCfg)

Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/190
Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/191
Depends-On: https://github.com/openstack-k8s-operators/mariadb-operator/pull/200

Jira: [OSPRH-4547](https://issues.redhat.com//browse/OSPRH-4547)

### CODE REVIEW

File: controllers/glance_controller.go [hunk 1]
```diff
@@ -588,12 +532,19 @@ func (r *GlanceReconciler) reconcileNormal(ctx context.Context, instance *glance
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 	// run check OpenStack secret - end
 
+	db, result, err := r.ensureDB(ctx, helper, instance)
```
Comment (fmount) [inline controllers/glance_controller.go:535]: @stuggi not sure if there's a reason to not overwrite the same `ctrlResult` variable here and use `result`

File: controllers/glance_controller.go [hunk 2]
```diff
@@ -1074,3 +1036,70 @@ func (r *GlanceReconciler) glanceAPICleanup(ctx context.Context, instance *glanc
 	}
 	return nil
 }
+
+func (r *GlanceReconciler) ensureDB(
```
Comment (fmount) [inline controllers/glance_controller.go:1040]: +1, good to have this logic into a `ensureDB` function

File: controllers/glance_controller.go [hunk 3]
```diff
@@ -919,7 +871,17 @@ func (r *GlanceReconciler) generateServiceConfig(
...
+	var tlsCfg *tls.Service
+	for _, api := range instance.Spec.GlanceAPIs {
+		if api.TLS.CaBundleSecretName != "" {
+			tlsCfg = &tls.Service{}
+			break
+		}
+	}
+	customData := map[string]string{
+		glance.CustomConfigFileName: instance.Spec.CustomServiceConfig,
+		"my.cnf":                    db.GetDatabaseClientConfig(tlsCfg), //(mschuppert) for now just get the default my.cnf
```
Comment (fmount) [inline controllers/glance_controller.go:883]: ack, so the whole thing here is that we can get `my.cnf` as part of an API provided by the mariadb db object. Then we can use this information to set the database connection string.

**Review by fmount — COMMENTED** [review]
Looks good to me.
We'll merge as long as it's rebased on main

File: controllers/glance_controller.go (hunk 1 above)
Comment (stuggi) [reply controllers/glance_controller.go:535]: no, not really a reason.

**Comment by stuggi** [comment]
need to update for the new added edge tests

File: controllers/glance_controller.go (hunk 1 above)
Comment (stuggi) [reply controllers/glance_controller.go]: changed to the ctrlResult

**Pull request merged**